### DISCUSSION TITLE
Exit at ending_year

### DISCUSSION THREAD

**Discussion by michalc** [OpenTTD/OpenTTD#11339] [Q&A] [answered]
Hello 👋 

It is possible for OpenTTD to exit at the ending_year? If this feature doesn't exist, where could it be added? Nosing around the codebase I see an `_exit_game` variable - I guess this would need to be set to `true` at the appropriate point?

Context: I'm trying to use OpenTTD not so much as a game, but as a sort of reproducible experiment framework - and so automatically exiting at a predefined point I _think_ wold be a required as part of this.

Hope you can help,

Michal

**Comment by Eddi-z** [comment, marked as answer]
the "null" video driver has a feature that lets it exit after running for X ticks:
`./openttd -vnull:ticks=10000`

**Reply by michalc** [reply]
Ah thank you!

**Comment by michalc** [comment]
Ah so it's maybe a bit of a hacky location... but I wonder if _exit_game could be set from here?

https://github.com/OpenTTD/OpenTTD/blob/beafdadde452b41042d5c552330d4418ce06ffe2/src/highscore_gui.cpp#L263C22-L263C22

```
if (TimerGameCalendar::year == _settings_game.game_creation.ending_year + 1) {
    ShowEndGameChart();
}
```
So instead of showing the and game chart, it exits
